### DISCUSSION TITLE
Feature request: hoisted multicall

### DISCUSSION THREAD

**Discussion by roninjin10** [wevm/viem#316] [Idea / Feature Request]
Feature request is simply to support hoisting multicall options.  Common use case is to be calling the same contract or same function.

For context this is how it works today:

```typescript
const results = await publicClient.multicall({
  contracts: [
    {
      ...wagmiContract,
      functionName: 'ownerOf',
      args: [6942069n]
    },
    {
      ...wagmiContract,
      functionName: 'ownerOf',
      args: [69420n]
    },
  ]
})
```

How I propose it could work is providing options higher hoists that option such that it applies to all calls

```typescript
const results = await publicClient.multicall({
   ...wagmiContract,
  functionName: 'ownerOf',
  contracts: [
    {
      args: [6942069n]
    },
    {
      args: [69420n]
    },
  ]
})
```

**Comment by hammeiam** [comment]
If this is just a syntactic change, would the following format be cleaner for you with the current state of viem?
```typescript
const callShared = {...wagmiContract, functionName: 'ownerOf'}
const results = await publicClient.multicall({
  contracts: nftIds.map(id => ({...callShared, args:[id]}))
})
```

**Reply by jxom** [reply]
Yeah, I'd prefer this approach. I can't say I am a fan of hoisting it up to the root as it gives the consumer the wrong idea of how multicall _actually_ works, plus, keeping the API surface area smaller reduces cognitive overload.

**Comment by roninjin10** [comment]
Yea coming back to this later it feels not worth the typescript complexity ect. don’t agree with past self that this is useful.
